xonsh/built_ins.py

```diff
@@ -345,7 +345,6 @@ def no_pg_xonsh_preexec_fn():
     """Default subprocess preexec function for when there is no existing
     pipeline group.
     """
-    os.setpgrp()
```

Comment:
Same as below, I think we want to keep this on linux, and perhaps add a comment that we don't know why it doesn't work on mac.